### PR TITLE
Use custom exception in classes 

### PR DESCRIPTION
**Pull request by JessickaMoya**

### CODE REVIEW

File: src/main/java/org/fundacion/jala/converter/core/facade/ExtractFacade.java [hunk 1]
```diff
@@ -56,10 +59,11 @@ public static void getTextExtract(final MultipartFile file, final String languag
      * @return a String with name of file which contains metadata.
      * @throws IOException is exception when invalid path.
      * @throws IllegalArgumentException is exception when string not correspond enum.
+     * @throws MetadataException if process is interrupted.
      */
     public static String getMetadataExtract(final MultipartFile file, final Boolean isMoreInfo,
                                             final String nameExport, final String format)
-            throws IOException, IllegalArgumentException {
+            throws IOException, IllegalArgumentException, MetadataException {
```
Comment (SaulCM) [inline src/main/java/org/fundacion/jala/converter/core/facade/ExtractFacade.java]: Different indentation, lines 65 and 67 should follow the same identation.

File: src/main/java/org/fundacion/jala/converter/controller/AudioConverterController.java [hunk 2]
```diff
@@ -64,7 +65,11 @@ public String uploadFile(final @RequestParam("file") MultipartFile file, final @
         String outputFilename = ConverterFacade.getAudioConverter(
                 new AudioParameter(parameterOutputChecksum.getOutputFilename(), format, bitrate, hz, volume,
                         audioChannel));
```
Comment (SaulCM) [inline src/main/java/org/fundacion/jala/converter/controller/AudioConverterController.java:67]: Lines 66 and 67 should have the same indentation.

File: src/main/java/org/fundacion/jala/converter/controller/VideoConverterController.java [hunk 3]
```diff
@@ -66,10 +67,19 @@ public String uploadFile(final @RequestParam("file") MultipartFile file,
...
         parameterOutputChecksum = ChecksumFacade.getChecksum(checksum, file);
-        String outputFilename = ConverterFacade.getVideoConverter(
-                new VideoParameter(parameterOutputChecksum.getOutputFilename(), outputFormat, resolution, thumbnail,
-                        frameRate, width, height, audio));
-        extractMetadata(metadata, outputFilename, fileStorageService);
+        String outputFilename = null;
+        try {
+            outputFilename = ConverterFacade.getVideoConverter(
+                    new VideoParameter(parameterOutputChecksum.getOutputFilename(), outputFormat, resolution, thumbnail,
+                            frameRate, width, height, audio));
```
Comment (SaulCM) [inline src/main/java/org/fundacion/jala/converter/controller/VideoConverterController.java:74]: Lines 73 and 74 should have the same indentation.

File: src/main/java/org/fundacion/jala/converter/core/ExtractText.java [hunk 4]
```diff
@@ -14,12 +14,11 @@
...
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
+import org.fundacion.jala.converter.core.exceptions.TextExtractorException;
 import org.fundacion.jala.converter.core.parameter.ExtractTextParameter;
 import org.fundacion.jala.converter.core.results.Result;
-import java.io.BufferedWriter;
-import java.io.File;
-import java.io.FileWriter;
-import java.io.PrintWriter;
+
```
Comment (SaulCM) [inline src/main/java/org/fundacion/jala/converter/core/ExtractText.java]: The empty line between imports should be removed.

File: src/main/java/org/fundacion/jala/converter/core/facade/CompilerFacade.java [hunk 5]
```diff
@@ -37,7 +34,7 @@ public CompilerFacade() {
      * @param code is string with code in java.
      * @return a String of the result on runtime console.
      */
-    public static String facadeJavaCompile(final String code) {
+    public static String facadeJavaCompile(final String code) throws CompilerException {
         if (!code.isBlank() || !code.equals(null)) {
             javaCompiler = new JavaCompiler();
             String filePath = Transform.toFile(code, "Main", "java");
```
Comment (SaulCM) [inline src/main/java/org/fundacion/jala/converter/core/facade/CompilerFacade.java:40]: "Main" and "Java" could be final attributes.

File: src/main/java/org/fundacion/jala/converter/core/facade/ExtractFacade.java (hunk 1 above)
Comment (SaulCM) [inline src/main/java/org/fundacion/jala/converter/core/facade/ExtractFacade.java]: Lines 65 and 66 should follow the same indentation.

File: src/main/java/org/fundacion/jala/converter/core/facade/MetadataFacade.java [hunk 6]
```diff
@@ -13,6 +13,8 @@
 
 import org.fundacion.jala.converter.core.ExtractMetadata;
 import org.fundacion.jala.converter.core.FileStorageService;
+import org.fundacion.jala.converter.core.exceptions.MetadataException;
+
```
Comment (SaulCM) [inline src/main/java/org/fundacion/jala/converter/core/facade/MetadataFacade.java]: The empty line between the imports should be removed.

File: src/main/java/org/fundacion/jala/converter/core/facade/ConverterFacade.java [hunk 7]
```diff
@@ -10,10 +10,12 @@
  */
 package org.fundacion.jala.converter.core.facade;
 
+import org.fundacion.jala.converter.core.exceptions.ConverterException;
 import org.fundacion.jala.converter.core.parameter.AudioParameter;
 import org.fundacion.jala.converter.core.parameter.VideoParameter;
 import org.fundacion.jala.converter.core.AudioConverter;
 import org.fundacion.jala.converter.core.VideoConverter;
+
```
Comment (edsonarios) [inline src/main/java/org/fundacion/jala/converter/core/facade/ConverterFacade.java]: This empty line should be deleted

File: src/main/java/org/fundacion/jala/converter/response/ErrorResponse.java [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+ */
+package org.fundacion.jala.converter.response;
+
+/**
+ * This class creates the application's responses.
+ */
+public class ErrorResponse extends PaoPaoResponse {
+    private String errorMessage;
+
+    public ErrorResponse(String initialStatus, String initialErrorMessage) {
```
Comment (edsonarios) [inline src/main/java/org/fundacion/jala/converter/response/ErrorResponse.java]: Parameters should be "final"

File: src/main/java/org/fundacion/jala/converter/response/ErrorResponse.java [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+    public String getErrorMessage() {
+        return errorMessage;
+    }
+
+    /**
+     * Sets the response error message.
+     *
+     * @param newErrorMessage is a String to change the error message.
+     */
+    public void setErrorMessage(String newErrorMessage) {
```
Comment (edsonarios) [inline src/main/java/org/fundacion/jala/converter/response/ErrorResponse.java]: Parameters should be "final"

File: src/main/java/org/fundacion/jala/converter/response/PaoPaoResponse.java [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+ */
+package org.fundacion.jala.converter.response;
+
+/**
+ * This class creates the application's responses.
+ */
+public class PaoPaoResponse {
+    private String status;
+
+    public PaoPaoResponse(String initialStatus) {
```
Comment (edsonarios) [inline src/main/java/org/fundacion/jala/converter/response/PaoPaoResponse.java]: Parameters should be "final"

File: src/main/java/org/fundacion/jala/converter/response/PaoPaoResponse.java [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+    public String getStatus() {
+        return status;
+    }
+
+    /**
+     * Sets the response status.
+     *
+     * @param newStatus is a String to change the status.
+     */
+    public void setStatus(String newStatus) {
```
Comment (edsonarios) [inline src/main/java/org/fundacion/jala/converter/response/PaoPaoResponse.java]: Parameters should be "final"

File: src/main/java/org/fundacion/jala/converter/core/facade/ConverterFacade.java (hunk 7 above)
Comment (jpgonzales27) [inline src/main/java/org/fundacion/jala/converter/core/facade/ConverterFacade.java]: jessi, this empty line should be removed

File: src/main/java/org/fundacion/jala/converter/core/facade/MetadataFacade.java [hunk 12]
```diff
@@ -28,7 +30,7 @@
      * @param outputFileName a String with the new file's name.
      * @param fileStorageService an object to create the path.
      */
-    public static void extractMetadata(final boolean metadata, final String outputFileName, final FileStorageService fileStorageService) {
+    public static void extractMetadata(final boolean metadata, final String outputFileName, final FileStorageService fileStorageService) throws MetadataException {
```
Comment (jpgonzales27) [inline src/main/java/org/fundacion/jala/converter/core/facade/MetadataFacade.java]: Jessi, Is this line not longer than 120 characters?

File: src/main/java/org/fundacion/jala/converter/core/facade/ExtractFacade.java (hunk 1 above)
Comment (JessickaMoya) [reply src/main/java/org/fundacion/jala/converter/core/facade/ExtractFacade.java]: Thanks, I have just fixed it

File: src/main/java/org/fundacion/jala/converter/controller/AudioConverterController.java (hunk 2 above)
Comment (JessickaMoya) [reply src/main/java/org/fundacion/jala/converter/controller/AudioConverterController.java:67]: The editor indentates it in that way

File: src/main/java/org/fundacion/jala/converter/controller/VideoConverterController.java (hunk 3 above)
Comment (JessickaMoya) [reply src/main/java/org/fundacion/jala/converter/controller/VideoConverterController.java:74]: The editor indentates it in that way

File: src/main/java/org/fundacion/jala/converter/core/ExtractText.java (hunk 4 above)
Comment (JessickaMoya) [reply src/main/java/org/fundacion/jala/converter/core/ExtractText.java]: Thanks, I have just fixed it

File: src/main/java/org/fundacion/jala/converter/core/facade/ExtractFacade.java (hunk 1 above)
Comment (JessickaMoya) [reply src/main/java/org/fundacion/jala/converter/core/facade/ExtractFacade.java]: Thanks, I have just fixed it

File: src/main/java/org/fundacion/jala/converter/core/facade/MetadataFacade.java (hunk 6 above)
Comment (JessickaMoya) [reply src/main/java/org/fundacion/jala/converter/core/facade/MetadataFacade.java]: Thanks, I have just fixed it

File: src/main/java/org/fundacion/jala/converter/core/facade/ConverterFacade.java (hunk 7 above)
Comment (JessickaMoya) [reply src/main/java/org/fundacion/jala/converter/core/facade/ConverterFacade.java]: Thanks, I have just fixed it

File: src/main/java/org/fundacion/jala/converter/response/ErrorResponse.java (hunk 8 above)
Comment (JessickaMoya) [reply src/main/java/org/fundacion/jala/converter/response/ErrorResponse.java]: Thanks, I have just fixed it

File: src/main/java/org/fundacion/jala/converter/response/ErrorResponse.java (hunk 9 above)
Comment (JessickaMoya) [reply src/main/java/org/fundacion/jala/converter/response/ErrorResponse.java]: Thanks, I have just fixed it

File: src/main/java/org/fundacion/jala/converter/response/PaoPaoResponse.java (hunk 10 above)
Comment (JessickaMoya) [reply src/main/java/org/fundacion/jala/converter/response/PaoPaoResponse.java]: Thanks, I have just fixed it

File: src/main/java/org/fundacion/jala/converter/response/PaoPaoResponse.java (hunk 11 above)
Comment (JessickaMoya) [reply src/main/java/org/fundacion/jala/converter/response/PaoPaoResponse.java]: Thanks, I have just fixed it

File: src/main/java/org/fundacion/jala/converter/core/facade/ConverterFacade.java (hunk 7 above)
Comment (JessickaMoya) [reply src/main/java/org/fundacion/jala/converter/core/facade/ConverterFacade.java]: Thanks, I have just fixed it

File: src/main/java/org/fundacion/jala/converter/core/facade/MetadataFacade.java (hunk 12 above)
Comment (JessickaMoya) [reply src/main/java/org/fundacion/jala/converter/core/facade/MetadataFacade.java]: Thanks, I have just fixed it

File: src/main/java/org/fundacion/jala/converter/controller/VideoConverterController.java (hunk 3 above)
Comment (SaulCM) [reply src/main/java/org/fundacion/jala/converter/controller/VideoConverterController.java:74]: It is good then.

File: src/main/java/org/fundacion/jala/converter/controller/AudioConverterController.java (hunk 2 above)
Comment (SaulCM) [reply src/main/java/org/fundacion/jala/converter/controller/AudioConverterController.java:67]: It is good then.

File: src/main/java/org/fundacion/jala/converter/core/facade/CompilerFacade.java (hunk 5 above)
Comment (JessickaMoya) [reply src/main/java/org/fundacion/jala/converter/core/facade/CompilerFacade.java:40]: Sorry, that's code I haven't done

**Review by SaulCM — APPROVED** [review]

**Review by jpgonzales27 — APPROVED** [review]

File: src/main/java/org/fundacion/jala/converter/core/ExtractText.java [hunk 13]
```diff
@@ -14,12 +14,10 @@
...
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
+import org.fundacion.jala.converter.core.exceptions.TextExtractorException;
 import org.fundacion.jala.converter.core.parameter.ExtractTextParameter;
 import org.fundacion.jala.converter.core.results.Result;
-import java.io.BufferedWriter;
-import java.io.File;
-import java.io.FileWriter;
-import java.io.PrintWriter;
+import java.io.*;
```
Comment (edsonarios) [inline src/main/java/org/fundacion/jala/converter/core/ExtractText.java]: The imports of the libraries should not be with "*"

Comment (JessickaMoya) [reply src/main/java/org/fundacion/jala/converter/core/ExtractText.java]: Thanks, I have just fixed it

**Review by edsonarios — APPROVED** [review]

**Pull request merged**